src/commonTest/kotlin/baaahs/net/TestNetwork.kt

```diff
@@ -61,6 +69,101 @@ class TestNetwork(var defaultMtu: Int = 1400) : Network {
         ): Network.TcpConnection {
             TODO("Link.connectWebSocket not implemented")
         }
+
+        private val mdns = TestMdns()
+
+        inner class TestMdns() : Network.Mdns {
+
+            override fun register(
+                hostname: String,
+                type: String,
+                proto: String,
+                port: Int,
+                domain: String,
+                params: MutableMap<String, String>
+            ): Network.MdnsRegisteredService? {
+                var dom = domain
+                if (dom.startsWith(".")) {
+                    dom = dom.substring(1)
+                }
+                if (!dom.endsWith(".")) {
+                    dom += "."
+                }
+                val fullname = "$hostname.$type.$proto.$domain"
+                val inst = TestRegisteredService(hostname, type, proto, port, domain, params)
+                services[fullname] = inst
+                (inst as? TestRegisteredService)?.announceResolved()
+                return inst
+            }
+
+            override fun unregister(inst: Network.MdnsRegisteredService?) {
+                if (inst != null) {
+                    val fullname = inst.hostname + "." + inst.type + "." + inst.proto + "." + inst.domain
+                    val removed = services.remove(fullname)
+                    if (removed != null) {
+                        (removed as? TestRegisteredService)?.announceRemoved()
+                    }
+                }
+            }
+
+            override fun listen(type: String, proto: String, domain: String, handler: Network.MdnsListenHandler) {
+                if (!listeners.containsKey("$type.$proto.$domain")) {
+                    listeners["$type.$proto.$domain"] = mutableListOf()
+                }
+                listeners["$type.$proto.$domain"]!!.add(handler)
+            }
+
+            open inner class TestMdnsService(override val hostname: String, override val type: String, override val proto: String, override val port: Int, override val domain: String, val params: MutableMap<String, String>) : Network.MdnsService {
+
+                private var id : Int = nextServiceId++
+
+                override fun getAddress(): Network.Address? {
+                    return Address("test-svc-$id")
+                }
+
+                override fun getTXT(key: String): String? {
+                    return params[key]
+                }
+
+                override fun getAllTXTs(): MutableMap<String, String> {
+                    return params
```

Comment:
For single line functions that return a value, use expression form (alt-enter, change to expression form)